fili-core/src/main/java/com/yahoo/bard/webservice/data/config/metric/makers/AggregationAverageMaker.java

```diff
@@ -115,80 +120,69 @@ private TemplateDruidQuery buildOuterQuery(
      * @return A template query representing the inner aggregation
      */
     private TemplateDruidQuery buildInnerQuery(MetricField sourceMetric, TemplateDruidQuery innerDependentQuery) {
-
         Set<Aggregation> newInnerAggregations = convertSketchesToSketchMerges(innerDependentQuery.getAggregations());
-        Set<PostAggregation> newInnerPostAggregations = new LinkedHashSet<>();
 
-        // If the sourceMetric is a Post Aggregator, we need to preserve it in the inner query
-        if (sourceMetric instanceof PostAggregation) {
-            newInnerPostAggregations.add((PostAggregation) sourceMetric);
-        }
+        Set<PostAggregation> newInnerPostAggregations = !(sourceMetric instanceof PostAggregation) ?
+                Collections.emptySet() : ImmutableSet.of((PostAggregation) sourceMetric);
 
         // Build the inner query with the new aggregations and with the count
-        TemplateDruidQuery innerQuery = innerDependentQuery;
-        innerQuery = innerQuery.withAggregations(newInnerAggregations);
-        innerQuery = innerQuery.withPostAggregations(newInnerPostAggregations);
-        innerQuery = innerQuery.merge(buildTimeGrainCounterQuery());
-        return innerQuery;
+        return innerDependentQuery.withAggregations(newInnerAggregations)
+                .withPostAggregations(newInnerPostAggregations)
+                .merge(buildTimeGrainCounterQuery());
     }
 
     /**
-     * If the aggregation being averaged is a sketch, it will become a sketch merge, and need to be estimated for
-     * summing to work at the outer level.
+     * If the aggregation being averaged is a sketch, the inner query must convert it to a numerical type so that it
+     * can be summed.
      *
      * @param originalSourceMetric  The metric being target for sums
      *
      * @return Either the original MetricField, or a new SketchEstimate post aggregation
      */
     private MetricField convertToSketchEstimateIfNeeded(MetricField originalSourceMetric) {
-        if (originalSourceMetric instanceof SketchAggregation) {
-            return FieldConverterSupplier.sketchConverter.asSketchEstimate((SketchAggregation) originalSourceMetric);
-        }
-        return originalSourceMetric;
+        return !(originalSourceMetric instanceof SketchAggregation) ?
+                originalSourceMetric :
+                sketchConverter.asSketchEstimate((SketchAggregation) originalSourceMetric);
```

Comment:
I would flip these conditions, and make the check a positive check rather than a negative one.